test/service/test_tdr.py

```diff
@@ -446,6 +446,12 @@ def _run_sql(self, query: str) -> BigQueryRows:
     def _full_table_name(self, table_name: str) -> str:
         return self._source.bq_name + '.' + table_name
 
+    @cached_property
```

Comment:
Let's subclass the TDRClient class instead. This approach doesn't scale so well if we need to mock/patch/customize other aspects of the client.